connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -63,6 +63,9 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[ExecutePlanResp
         }
 
       val executeHolder = sessionHolder.createExecutePlanHolder(v)
+
+      sessionHolder.events.postStarted(executeHolder)
```

Comment:
Note that in https://github.com/apache/spark/pull/41315 (WIP), I am moving the actual execution to a different thread, kind of like Thriftserver starts the actual execution in SparkExecuteOperation.runInternal -> execute.
Do we want to consider having separate STARTED and EXECUTING/RUNNING, the way thriftserver has?

Reply:
... or after that PR consider it PENDING until the inner thread starts running.

Reply:
Did Thriftserver expose pending / running events?
Will requests be put in a queue before being executed on the separate thread? Do we expect the time between receiving the request and processing it to be significant?

A few thoughts:
1) It would be preferable for both implementation to align as events with the same name are meant to have the similar meaning.
2) Unless the request is queued, from a user perspective, it seems like the request has started processing and the threading is an implementation detail.
3) We should measure time from end to end as much as possible.